### PR TITLE
sigi 3.6.3 (new formula)

### PR DESCRIPTION
**Pull request by booniepepper**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I'm hoping to (finally) promote the tap at https://github.com/sigi-cli/homebrew-sigi into homebrew-core

[Packaging status](https://repology.org/project/sigi/versions)

A majority of users are installing from cargo directly (6k downloads https://crates.io/crates/sigi)

Not many users of Sigi engage with the GitHub repo

### CODE REVIEW

File: Formula/s/sigi.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  def install
+    # When using --head, Homebrew does a git clone and the install method
+    # is run from the context of that directory.
+    unless build.head?
+      # Otherwise, the create file is downloaded to a temporary directory, and
+      # then the install method is run from that directory, so we need to
+      # extract that crate file first. A crate file is just a tarball with a
+      # different file extension.
+      system "tar", "--strip-components", "1", "-xzvf", "sigi-#{version}.crate"
+    end
```
Comment (SMillerDev) [inline Formula/s/sigi.rb:20]:
```suggestion
    if build.stable?
      system "tar", "--strip-components", "1", "-xzvf", "sigi-#{version}.crate"
    end
```

**Comment by booniepepper** [comment]
> GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)

Yikes, is there a way to bypass the "notable" test here?

Not all users of Sigi are going to the github repo and interacting with it. Many users are not developers. A majority are just installing from cargo directly (6k downloads https://crates.io/crates/sigi) and some others are getting it from Linux repos

image

File: Formula/s/sigi.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    system "cargo", "install", *std_cargo_args
+    bin.install "target/release/sigi"
+    man1.install "sigi.1"
+  end
+
+  test do
+    system "#{bin}/sigi", "--version"
```
[the remark this answers is not in the document]
Comment (booniepepper) [reply Formula/s/sigi.rb]: Thanks for taking a look! I've updated the tests

File: Formula/s/sigi.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Sigi < Formula
+  desc "Organizing tool for terminal lovers that hate organizing"
+  homepage "https://sigi-cli.org"
+  url "https://static.crates.io/crates/sigi/sigi-3.6.3.crate"
+  sha256 "15f5ebbe7e51bdc3236f69371805deebbbdd730519cc8e680ba78206b668d96d"
```
Comment (chenrui333) [inline Formula/s/sigi.rb:5]:
```suggestion
  url "https://github.com/sigi-cli/sigi/archive/refs/tags/v3.6.3.tar.gz"
  sha256 "8bd08b8fb372cf0bcefd45275f8a6956d0f551e9c940c8724f55770ef5b79612"
```

File: Formula/s/sigi.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  license "GPL-2.0-only"
+  head "https://github.com/sigi-cli/sigi.git", branch: "core"
+
+  depends_on "rust" => :build
+
+  def install
+    system "tar", "--strip-components", "1", "-xzvf", "sigi-#{version}.crate" if build.stable?
+
```
Comment (chenrui333) [inline Formula/s/sigi.rb:13]: no need to have this if build from github source tarball
```suggestion
```

File: Formula/s/sigi.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    system "cargo", "install", *std_cargo_args
+    bin.install "target/release/sigi"
```
Comment (chenrui333) [inline Formula/s/sigi.rb:15]: this should be covered by L14
```suggestion
```

**Comment by chenrui333** [comment]
> > GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
> 
> Yikes, is there a way to bypass the "notable" test here?
> 
> Not many users of Sigi are going to go to the github repo and interact with it. Many users are not developers. A majority are just installing from cargo directly (6k downloads https://crates.io/crates/sigi) and some others are getting it from Linux repos
> 
> image

@booniepepper also you can move the package status into the OP.

**Comment by booniepepper** [comment]
@chenrui333 does homebrew prefer pre-built binaries over source builds? I'm OK making the switch

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@booniepepper, thanks for your first contribution to homebrew-core! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Comment by booniepepper** [comment]
@chenrui333 is there something I can do here? I don't really understand the bottling failure here

**Comment by chenrui333** [comment]
that failure is not related to notability, mostly a brew issue, need a master branch rebase

```
homebrew/core
  * The following should either be an alias or a rename, not both: bigdata, boot2docker, camlistore, crystal-lang, findbugs, gitlab-ci-multi-runner, gnome-icon-theme, gnupg2, gtef, gtk+4, htop-osx, latexila, letsencrypt, libmongoclient, linux-headers, mat, mobile-shell, mongo-c, newsbeuter, osh, pyqt5, qt5, recipes, speedtest_cli, ssreflect, tachyon, tensorflow and transfig
```

**Comment by chenrui333** [comment]
@booniepepper can you run a rebase to the latest master and forced update your branch? Thanks!

**Comment by booniepepper** [comment]
@chenrui333 Thanks, rebased!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @booniepepper! 🙌

**Pull request merged**